cmd/opentelemetry/app/exporter/elasticsearchexporter/esmodeltranslator/modeltranslator.go

```diff
@@ -222,21 +222,15 @@ func references(links pdata.SpanLinkSlice, parentSpanID pdata.SpanID, traceID db
 }
 
 func convertSpanID(spanID pdata.SpanID) (dbmodel.SpanID, error) {
-	spanIDInt, err := tracetranslator.BytesToUInt64SpanID(spanID.Bytes())
-	if err != nil {
-		return "", err
-	}
+	spanIDInt := tracetranslator.BytesToUInt64SpanID(spanID.Bytes())
 	if spanIDInt == 0 {
 		return "", errZeroSpanID
```

Comment:
This feels misplaced to me. This kind of validation should've been done way earlier in the pipeline. The ES backend itself doesn't actually care if IDs are all zeroes, it would still work.

Reply:
You're definitely right, but practically having this error handled here can prevent some time of unnecessary debugging in case of zero IDs and missing validation earlier. I believe this error arises systematically, so anyway there will be collisions by ID and this will only implicitly indicate zero ID error.